dlt/destinations/impl/redshift/redshift.py

```diff
@@ -65,25 +65,27 @@ def __init__(
         file_path: str,
         staging_credentials: Optional[CredentialsConfiguration] = None,
         staging_iam_role: str = None,
+        s3_region: str = "us-east-1",  # Add region as a parameter
     ) -> None:
         super().__init__(file_path, staging_credentials)
         self._staging_iam_role = staging_iam_role
+        self._s3_region = s3_region  # Store region
         self._job_client: "RedshiftClient" = None
 
     def run(self) -> None:
         self._sql_client = self._job_client.sql_client
-        # we assume s3 credentials where provided for the staging
+        # we assume S3 credentials were provided for the staging
         credentials = ""
         if self._staging_iam_role:
-            credentials = f"IAM_ROLE '{self._staging_iam_role}'"
+            credentials = f"IAM_ROLE '{self._staging_iam_role}' REGION '{self._s3_region}'"
         elif self._staging_credentials and isinstance(
             self._staging_credentials, AwsCredentialsWithoutDefaults
         ):
             aws_access_key = self._staging_credentials.aws_access_key_id
             aws_secret_key = self._staging_credentials.aws_secret_access_key
             credentials = (
                 "CREDENTIALS"
-                f" 'aws_access_key_id={aws_access_key};aws_secret_access_key={aws_secret_key}'"
+                f" 'aws_access_key_id={aws_access_key};aws_secret_access_key={aws_secret_key};region={self._s3_region}'"
```

Comment:
question: the AwsCredentialsWithoutDefaults already has an attribute `region_name`. shouldnt we take that one then?

Reply:
yeah you are right.
I changed it and we can now use region from the AwsCredentialsWithoutDefaults if it is provisioned :)